.ci/do_release.sh

```diff
@@ -1,77 +1,72 @@
-    echo "FULL_VERSION from script parameter: [${release_version}]"
+if [ -z "${release_tag}" ]; then
+    echo "ERROR: 'release_tag' was not set. Exit"
+    exit 1
+fi
+
+if [ -z "${revision}" ]; then
+    echo "WARN: 'revision' was not set, defaulting to 1"
+    revision=1
 fi
 
-env PRJ_RELEASE=1 contrib/build_pkg.sh -s
+if [ -z "${do_release}" ]; then
+    echo "WARN: 'do_release' was not set, defaulting to false (package will not be release)"
+    do_release=false
+fi
+
+env PRJ_RELEASE="${revision}" contrib/build_pkg.sh -s
 
-MAJOR_VERSION=$(cat configure.ac | grep -e "define(\[prj_ver_major\]" | awk -e '{ printf $2 };' | sed  's/)//g')
-MINOR_VERSION=$(cat configure.ac | grep -e "define(\[prj_ver_minor\]" | awk -e '{ printf $2 };' | sed  's/)//g')
-REVISION_VERSION=$(cat configure.ac | grep -e "define(\[prj_ver_revision\]" | awk -e '{ printf $2 };' | sed  's/)//g')
+MAJOR_VERSION=$(grep -e "define(\[prj_ver_major\]" configure.ac | awk -e '{ printf $2 };' | sed  's/)//g')
+MINOR_VERSION=$(grep -e "define(\[prj_ver_minor\]" configure.ac | awk -e '{ printf $2 };' | sed  's/)//g')
+REVISION_VERSION=$(grep -e "define(\[prj_ver_revision\]" configure.ac | awk -e '{ printf $2 };' | sed  's/)//g')
 configure_ac_version="${MAJOR_VERSION}.${MINOR_VERSION}.${REVISION_VERSION}"
 echo "FULL_VERSION from configure.ac: [${configure_ac_version}]"
 
-last_tag=$(git describe --tags $(git rev-list --tags --max-count=1))
-echo "Last tag: [${last_tag}]"
-
-if [[ "$last_tag" != "${configure_ac_version}" ]]; then
-    echo "ERROR: FULL_VERSION from configure.ac doesn't match last tag version! Exit"
+if [[ "${release_tag}" != "${configure_ac_version}" ]]; then
+    echo "ERROR: FULL_VERSION from configure.ac doesn't match tag provided! Exit"
     exit 1
 fi
 
-if [ -z "${release_version}" ]; then
-    release_version=${configure_ac_version}
-else
-    if [[ "$last_tag" != "${release_version}" ]]; then
-        echo "ERROR: FULL_VERSION from script parameter doesn't match last tag version! Exit"
-        exit 1
-    fi
-fi
-
-_UID=6213
-_GID=101
-_LOGIN=swx-jenkins
-_HOME=/var/home/$_LOGIN
-echo "${_LOGIN} ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
-echo "root ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
-groupadd -f -g "$_GID" "$_LOGIN"
-useradd -u "$_UID" -g "$_GID" -s /bin/bash -m -d ${_HOME} "${_LOGIN}"
+if [ "${do_release}" = true ] ; then
+    echo "do_release is set to true, will release package into ${release_folder}/${release_tag}"
 
-cd $WORKSPACE/pkg/packages
+    cd pkg/packages || { echo "pkg folder is missing, exiting..."; exit 1; }
+    pkg_name=$(ls -1 libxlio-*.src.rpm)
+    DST_DIR=${release_folder}/${release_tag}
 
-pkg_name=$(ls -1 libxlio-*.src.rpm)
+    if [[ -e "${DST_DIR}/${pkg_name}" ]]; then 
+        echo "ERROR: [${DST_DIR}/${pkg_name}] file already exist. Exit"
+        exit 1
+    fi
 
-DST_DIR=${release_folder}/${release_version}
-sudo -E -u swx-jenkins sh -c "mkdir -p $DST_DIR"
+    sudo -E -u swx-jenkins sh -c "mkdir -p $DST_DIR"
```

Comment:
Why are you using the `sh` here? `sudo` can execute command with parameters without any additional hacks

Reply:
If you insist on using the `sh -c "something something"` combination why `sh` and not `bash`

`sh` != `bash`